packages/react-components/react-search/src/components/SearchBox/useSearchBox.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { resolveShorthand, useControllableState, useEventCallback } from '@fluentui/react-utilities';
+import { Input } from '@fluentui/react-input';
+import type { SearchBoxProps, SearchBoxState } from './SearchBox.types';
+import { DismissRegular, SearchRegular } from '@fluentui/react-icons';
+
+/**
+ * Create the state required to render SearchBox.
+ *
+ * The returned state can be modified with hooks such as useSearchBoxStyles_unstable,
+ * before being passed to renderSearchBox_unstable.
+ *
+ * @param props - props from this instance of SearchBox
+ * @param ref - reference to root HTMLElement of SearchBox
+ */
+export const useSearchBox_unstable = (props: SearchBoxProps, ref: React.Ref<HTMLElement>): SearchBoxState => {
+  const { contentBefore, dismiss, contentAfter, ...inputProps } = props;
+
+  const [value, setValue] = useControllableState({
+    state: props.value,
+    defaultState: props.defaultValue,
+    initialState: '',
+  });
+
+  const onDismissClick = React.useCallback(
+    ev => {
+      setValue('');
+    },
+    [setValue],
+  );
+
+  const state: SearchBoxState = {
+    components: {
+      root: Input,
+      contentBefore: 'span',
+      dismiss: 'span',
+      contentAfter: 'span',
+    },
+
+    root: {
+      type: 'search',
+      input: {}, // defining here to have access in styles hook
+      value,
+      onChange: useEventCallback(ev => {
+        const newValue = ev.target.value;
+        setValue(newValue);
+      }),
+      ...inputProps,
+    },
+    contentBefore: resolveShorthand(contentBefore, {
+      defaultProps: {
+        children: <SearchRegular />,
+      },
+      required: true, // TODO need to allow users to remove
+    }),
+    dismiss: resolveShorthand(dismiss, {
+      defaultProps: {
+        children: <DismissRegular onClick={onDismissClick} />,
```

Comment:
the `onClick` should be on the slot itself rather than the child icon -- that way if an author replaces the icon with another one, they don't lose the click event.

A couple other things to add are `role: button` and `aria-label: clear` for voice control and touch screen reader users (it doesn't need to be focusable, since keyboard users can just do ctrl + a and backspace)

Reply:
Done, though are we intending to allow users to replace the dismiss icon?